pkg/virtual/framework/types.go

```diff
@@ -45,6 +46,7 @@ type ReadyFunc func() error
 type VirtualWorkspace interface {
 	GetName() string
 	ResolveRootPath(urlPath string, context context.Context) (accepted bool, prefixToStrip string, completedContext context.Context)
+	Authorize(ctx context.Context, a authorizer.Attributes) (authorized authorizer.Decision, reason string, err error)
```

Comment:
Should we embed the authorizer interface?

Reply:
Good idea